tower-util/src/map/layer.rs

```diff
@@ -0,0 +1,26 @@
+use super::MapRequest;
+use tower_layer::Layer;
+
+/// Map inbound requests to a new type
+#[derive(Debug, Clone)]
+pub struct MapRequestLayer<M> {
+  m: M,
+}
+
+impl<M> MapRequestLayer<M> {
+  /// Create a new `MapRequestLayer` from a closure
+  pub fn new(m: M) -> Self {
+    MapRequestLayer { m }
+  }
+}
+
+impl<S, M> Layer<S> for MapRequestLayer<M>
+where
+  M: Clone,
```

Comment:
I wonder if boxing is better here?

Reply:
@LucioFranco i'll leave this question for wiser Rust people than myself. Happy to do what y'all think is best. Working on adding the custom Fn trait now. :)